### DISCUSSION TITLE
zIndex для Popup

### DISCUSSION THREAD

**Discussion by FreeZerovshik** [yandex-maps-unofficial/vue-yandex-maps#242] [Q&A (2.x)]
При открытие popup остальные маркеры перекрывают. Пробовал различные варианты с zIndex не помогает (

![image](https://github.com/yandex-maps-unofficial/vue-yandex-maps/assets/1534775/808797bf-a478-45e5-bb10-66eefb865bd1)

**Comment by daniluk4000** [comment]
Добрый день. Вы можете задавать z-index в settings yandex-map-marker. По умолчанию Яндекс им всем 0 ставит, он перекрывает всё остальное, что вы делаете.

Если это не поможет, то от вас потребуется код, без него помочь будет трудновато. Но думаю это должно исправить ситуацию.

**Comment by FreeZerovshik** [comment]
Добрый день!
попробовал менять z-index в settings, без результата. 
В dev  консоле видно, что индекс изменился но маркеры также перекрывают.

код для работы с popup взял из примера в документации

<details>

<summary>Код блока с картой</summary>

<template>
  <yandex-map
    :v-model="map"
    :settings="{
      location: {
        center: [37.617644, 55.755819],
        zoom: 9,
      },
    }"
    width="100%"
    height="100%"
  >
    <yandex-map-default-scheme-layer />
    <yandex-map-default-features-layer />
    <yandex-map-controls :settings="{ position: 'right' }">
      <yandex-map-zoom-control />
    </yandex-map-controls>

    <template v-for="(marker, idx) in modelValue">
      <yandex-map-default-marker
        :settings="{
          coordinates: marker.coords,
          title: marker.name,

          color: 'blue',
          popup: { position: 'top', hidesMarker: true },
          zIndex: -1,
        }"
      >
        <template #popup="{ close }">
          <div class="marker-popup" @click="close()">
            <UPlaceBallon v-model="marker.name" :place="marker" />
          </div>
        </template>
      </yandex-map-default-marker>
    </template>
  </yandex-map>
</template>

<style>
.marker-popup {
  background: #fff;
  border-radius: 10px;
  padding: 10px;
  color: black;
  cursor: pointer;
  font-size: 14px;
  white-space: nowrap;
  height: 26rem;
  width: 45rem;
}

.marker {
  background: green;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 10px;
  cursor: pointer;
  color: #fff;
  padding: 10px 20px;
  white-space: nowrap;
}

.popup {
  position: absolute;
  top: calc(100% + 10px);
  background: #fff;
  border-radius: 10px;
  padding: 10px;
  color: black;
}
</style>

</details>

<details>
<summary>Код блока для popup</summary>

<template>
  <div class="p-2" v-if="place?.type">
    <div class="grid nested grid">
      <div class="col-12">
        <div class="grid">
          <div class="col-10 p-0">
            <h2 class="text-xl font-bold">
              {{ place?.additionalType + ' ' + place.name }}
            </h2>
          </div>
          <div
            class="col-2 flex flex-wrap justify-content-end align-items-center"
            v-if="place?.placeRate?.avgRate > 1"
          >
            <Avatar
              :label="place?.placeRate?.avgRate.toPrecision(2)"
              size="large"
              :style="calcRateClass(place?.placeRate?.avgRate)"
            />
          </div>

          <div class="col-10 -mt-2">
            <i class="text-ml pi pi-map-marker mr-2" style="color: #0074d9"></i>
            <a class="text-sm">{{ place?.shortAddress }}</a>
          </div>
          <div class="col-2 -mt-2">
            <div class="flex flex-row-reverse flex-wrap" v-if="place?.views">
              <div class="flex align-items-center justify-content-center">
                <span
                  class="material-symbols-outlined text-base text-400"
                  v-tooltip="'Количество просмотров'"
                >
                  visibility
                </span>
                <span class="text-sm text-400 ml-1"> {{ place?.views }}</span>
              </div>
              <div class="flex mr-2" v-if="place?.commentCnt">
                <span
                  class="material-symbols-outlined text-base mt-1 text-400"
                  v-tooltip="'Количество комментариев'"
                >
                  comment
                </span>
                <span class="flex text-sm text-400 align-items-center gap-2 ml-2">
                  {{ place?.commentCnt }}
                </span>
              </div>
            </div>
          </div>
        </div>
        <Divider layout="horizontal" class="m-0" />
      </div>
    </div>
    <div class="grid">
      <div class="col-5 flex flex-wrap justify-content-center align-items-center">
        <div v-if="place?.mainPictThumbnail" class="flex">
          <img
            class="shadow-2 max-w-full"
            :src="`${place?.mainPictThumbnail}`"
            :alt="place?.name"
            style="height: 12rem; width: auto"
          />
        </div>
        <div v-else>
          <div class="flex align-items-left justify-content-center">Изображение отсутствует</div>
        </div>
      </div>
      <div class="col-6">
        <div
          class="flex flex-row sm:flex-row justify-content-between align-items-center xl:align-items-start"
          style="height: 12rem"
        >
          <div class="flex align-items-center sm:align-items-start font-light text-base">
            <div class="flex flex-column gap-2">
              <div class="flex" v-if="place?.properties?.seasonTypes.length">
                <span class="material-symbols-outlined text-xl"> thermostat </span>
                <span
                  class="flex text-sm align-items-center gap-2 ml-2"
                  v-for="tg of place?.properties?.seasonTypes"
                  :key="tg"
                >
                  <span v-tooltip="tg?.description">{{ tg.name }}</span>
                </span>
              </div>
              <div class="flex" v-if="place?.properties?.relaxTypes.length">
                <span class="material-symbols-outlined text-xl"> emoji_people </span>

                <span
                  v-for="tg of place?.properties?.relaxTypes"
                  :key="tg"
                  class="flex text-sm align-items-center gap-2 ml-2"
                  ><span v-tooltip="tg?.description">{{ tg?.name }}</span></span
                >
              </div>
              <div class="flex" v-if="place.properties?.lifeTypes.length">
                <span class="material-symbols-outlined text-xl"> home </span>

                <span
                  v-for="tg of place?.properties?.lifeTypes"
                  :key="tg"
                  class="flex text-sm align-items-center gap-2 ml-2"
                  ><span v-tooltip="tg?.description">{{ tg?.name }}</span></span
                >
              </div>
              <div class="flex" v-if="place?.properties?.foodTypes.length">
                <span class="material-symbols-outlined text-xl"> restaurant </span>

                <span
                  v-for="tg of place?.properties?.foodTypes"
                  :key="tg"
                  class="flex text-sm align-items-center gap-2 ml-2"
                  ><span v-tooltip="tg?.description">{{ tg?.name }}</span></span
                >
              </div>
              <div class="flex" v-if="place?.properties?.catering">
                <span class="material-symbols-outlined text-xl"> room_service </span>
                <span
                  class="flex text-sm align-items-center gap-2 ml-2"
                  v-tooltip="'Предоставление услуг выездного обслуживания'"
                >
                  {{ place?.properties?.catering ? 'Да' : 'Нет' }}
                </span>
              </div>
              <div class="flex" v-if="place?.properties?.bankets">
                <span class="material-symbols-outlined text-xl"> brunch_dining </span>
                <span
                  class="flex text-sm align-items-center gap-2 ml-2"
                  v-tooltip="'Предоставление улус по проведению банкетов'"
                >
                  {{ place?.properties?.bankets ? 'Да' : 'Нет' }}
                </span>
              </div>
              <div class="flex" v-if="place?.properties?.minPrice">
                <span class="material-symbols-outlined text-xl"> sell </span>
                <span
                  class="flex text-sm align-items-center gap-2 ml-2"
                  v-tooltip="'Минимальная стоимость'"
                >
                  {{ 'от ' + place?.properties?.minPrice + ' руб./сутки' }}
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>

      <div class="col-12 h-2rem">
        <Divider layout="horizontal m-0" v-if="place?.properties?.featureTypes.length" />
        <div class="flex flex-wrap">
          <div class="flex" v-for="tg of place?.properties?.featureTypes" :key="tg.id">
            <Tag
              :value="tg.name"
              class="mr-1 mt-2 text-xs flex"
              style="background: #0074d9"
              v-tooltip="tg?.description"
            ></Tag>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
</details>

**Reply by daniluk4000** [reply]
@FreeZerovshik чтобы регулировать приоритеты, нужно zIndex менять на основании, того получается, открыт ли popup

А как это делать - вопрос хороший. Странно, что в самом Яндексе не сделали. Я вам советую переписать этот код на yandex-map-marker и самостоятельно накодить и маркер, и попап. Я так посмотрел, выглядит так, как будто в дефолтном это не продумано на стороне Яндекса

**Reply by FreeZerovshik** [reply]
спасибо, попробую
